ingest/src/test/java/com/microsoft/azure/kusto/ingest/QueuedIngestClientTest.java

```diff
@@ -133,12 +134,29 @@ void IngestFromBlob_NullBlobSourceInfo_IllegalArgumentException() {
     void IngestFromBlob_IngestionReportMethodIsTable_RemovesSecrets() throws Exception {
         BlobSourceInfo blobSourceInfo = new BlobSourceInfo("https://storage.table.core.windows.net/ingestionsstatus20190505?sv=2018-03-28&tn=ingestionsstatus20190505&sig=anAusomeSecret%2FK024xNydFzT%2B2cCE%2BA2S8Y6U%3D&st=2019-05-05T09%3A00%3A31Z&se=2019-05-09T10%3A00%3A31Z&sp=raud", 100);
         ingestionProperties.setReportMethod(IngestionProperties.IngestionReportMethod.TABLE);
-        ArgumentCaptor<TableServiceEntity> captur = ArgumentCaptor.forClass(TableServiceEntity.class);
+        ArgumentCaptor<TableServiceEntity> captor = ArgumentCaptor.forClass(TableServiceEntity.class);
 
         queuedIngestClient.ingestFromBlob(blobSourceInfo, ingestionProperties);
 
-        verify(azureStorageClientMock, atLeast(1)).azureTableInsertEntity(anyString(), captur.capture());
-        assert (((IngestionStatus) captur.getValue()).getIngestionSourcePath()).equals("https://storage.table.core.windows.net/ingestionsstatus20190505");
+        verify(azureStorageClientMock, atLeast(1)).azureTableInsertEntity(anyString(), captor.capture());
+        assert (((IngestionStatus) captor.getValue()).getIngestionSourcePath()).equals("https://storage.table.core.windows.net/ingestionsstatus20190505");
+    }
+
+    @Test
+    void IngestFromBlob_IngestionIgnoreFirstRecord_SetsProperty() throws Exception {
+        BlobSourceInfo blobSourceInfo = new BlobSourceInfo("https://storage.table.core.windows.net/ingestionsstatus20190505?sv=2018-03-28&tn=ingestionsstatus20190505&sig=anAusomeSecret%2FK024xNydFzT%2B2cCE%2BA2S8Y6U%3D&st=2019-05-05T09%3A00%3A31Z&se=2019-05-09T10%3A00%3A31Z&sp=raud", 100);
+        ingestionProperties.setIgnoreFirstRecord(true);
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+
+        queuedIngestClient.ingestFromBlob(blobSourceInfo, ingestionProperties);
+
+        verify(azureStorageClientMock, atLeast(1)).postMessageToQueue(anyString(), captor.capture());
+        assertTrue ((captor.getValue()).contains("\"ignoreFirstRecord\":\"true\""));
+
+        ingestionProperties.setIgnoreFirstRecord(false);
+        queuedIngestClient.ingestFromBlob(blobSourceInfo, ingestionProperties);
+        verify(azureStorageClientMock, atLeast(1)).postMessageToQueue(anyString(), captor.capture());
+        assertTrue ((captor.getValue()).contains("\"ignoreFirstRecord\":\"false\""));
```

Comment:
My approach is to prefer balance between UTs for each unit, and sometimes logically grouping UTs. In this case, I test both setting the field to true and false. (I could also check when the field isn't set at all, but I don't see much of a point.)

Reply:
Maybe add an e2e test to make sure the parameter we send actually does someting?

Reply:
You could say the same about testing with ingestion maps, but these tests already caught issues with how we send the parameters (in node), so I wouldn't say it's oos

Reply:
Just to be clear, we're talking about tests that fail if and only if there is a bug in the external service (in this case, the Kusto service).
These tests you reference that helped you catch a bug - the bug was in the external service?

I maintain that **on principle** such tests should never be included. It then becomes a slippery and weakens the service's responsibility to properly test itself.

Reply:
The bug wasn't on the external service, it was with how we sent the data.
An e2e test ensures that we have used the parameter correctly from the client's side.

Reply:
The UTs I added verify that the Java SDK code sends the property when it should.

And I verified that this property as sent by the Java SDK is appropriate as expected by the service, and has the expected impact on the service (per https://docs.microsoft.com/azure/data-explorer/ingestion-properties).

But once we know that the property we send is appropriate, retesting that during every build/CI means to me that we're testing the service, not the SDK. It's that principle, less than the fact that E2Es are expensive, that makes me uneasy with adding a Kusto service test to the SDK.